polyfill/lib/calendar.mjs

```diff
@@ -352,11 +396,32 @@ const jpn = {
     return idx - 1;
   },
 
-  isoYear(year, era) {
+  validateFields(fields) {
+    if (fields.eraYear === undefined && fields.year === undefined) {
+      throw new TypeError(
+        `required properties missing or undefined: must include 'year' and/or both 'era' and 'eraYear'`
+      );
+    }
+    if (fields.year === undefined || fields.eraYear === undefined) {
+      return;
```

Comment:
Isn't there more validation needed?  For example, eraYear should always be positive except if the era is the oldest one we know about.

Reply:
Ah, you're absolutely right. Quite sloppy, was just trying to make the tests pass. Will update.